src/Config/MosaicConfigV2.ts

```diff
@@ -0,0 +1,89 @@
+import * as path from "path";
+import Directory from "../Directory";
+import Logger from "../Logger";
+import * as fs from "fs-extra";
+
+/**
+ * Holds the config of mosaic chains of a specific origin chain.
+ */
+export default class MosaicConfig {
+    public originChain: OriginChain;
+    public auxiliaryChains: AuxiliaryChain[];
+
+    /**
+     * Saves this config to a file in its auxiliary chain directory.
+     */
+    public writeToMosaicConfigDirectory(): void {
+        const configPath = path.join(
+            Directory.getProjectMosaicConfigDir(),
+            `${this.originChain.chain}.json`,
+        );
+        Logger.info('storing mosaic config', {configPath});
+
+        fs.writeFileSync(
+            configPath,
+            JSON.stringify(this, null, '    '),
+        );
+    }
+}
+
+/**
+ * Holds origin chain specific config.
+ */
+class OriginChain {
+    public chain: string;
+    public contractAddress: OriginLibraries;
+}
+
+/**
+ * Holds config of a auxiliary chain.
+ */
+class AuxiliaryChain {
+    public chainId: string;
+    public bootNodes: string[];
+    public genesis: Object;
+    public contractAddress: ContractAddresses;
+}
+
+/**
+ * Hold contract addresses on origin and auxiliary chain specific to a auxiliary chain.
+ */
+class ContractAddresses {
+    public origin: OriginContract;
+    public auxiliary: AuxiliaryContract;
+}
+
+/**
+ * Hold contract addresses on origin chain independent of auxiliary chain.
+ */
+class OriginLibraries {
+    public simpleTokenAddress: string;
+    public merklePatricialLibAddress: string;
+    public gatewayLibAddress: string;
+    public messageBusAddress: string;
+}
+
+/**
+ * Contract addresses on the origin chain specific to a auxiliary chain.
+ */
+class OriginContract {
+    public anchorOrganizationAddress: string;
```

Comment:
We could add a new `type Address = string`. I think that would actually help a lot across the repo. You can define it in its own file and we can start using it here in the new config.